libraries/botbuilder-ai/src/luisRecognizer.ts

```diff
@@ -43,40 +43,198 @@ export class LuisRecognizer extends IntentRecognizer {
             this.options = { appId: appId, subscriptionKey: subscriptionKey as string };
         } else {
             this.options = Object.assign({}, appId);
+        }   
+
+        // We need to set verbose to true in order to get all intents
+        if(this.options.options) {
+            this.options.options.verbose = true;
+        }
+        else {
+            this.options.options = { verbose : true };
         }
 
         // Create client and override callbacks
+        let $this = this;
         const baseUri = (this.options.serviceEndpoint || 'https://westus.api.cognitive.microsoft.com');
         this.luisClient = new LuisClient(baseUri + '/luis/');
         this.onRecognize((context) => {
-            const intents: Intent[] = [];
             const utterance = (context.request.text || '').trim();
-            return LuisRecognizer.recognizeAndMap(this.luisClient, utterance, this.options)
-                .then(res => {
-                    intents.push(res);
-                    return intents;
-                });
+            return $this.recognizeAndMap( utterance,true).then(res =>{
+                let recognizerResults : RecognizerResult[]  = [res]
+                return recognizerResults;
+            });
         });
     }
 
-    public static recognize(utterance: string, options: LuisRecognizerOptions): Promise<Intent> {
-        const baseUri = (options.serviceEndpoint || 'https://westus.api.cognitive.microsoft.com');
-        var client = new LuisClient(baseUri + '/luis/');
-        return LuisRecognizer.recognizeAndMap(client, utterance, options);
+    public static recognize(utterance: string, options: LuisRecognizerOptions): Promise<RecognizerResult> {
+        let recognizer = new LuisRecognizer(options);
+        return recognizer.recognizeAndMap(utterance, true);
+    }
+
+    protected recognizeAndMap(utterance: string, verbose: boolean): Promise<RecognizerResult> {
+        let $this = this;
+        return this.luisClient.getIntentsAndEntitiesV2($this.options.appId, this.options.subscriptionKey, utterance, $this.options.options)
+            .then((result : LuisResult) => {
+                let entitiesAndMetadata = $this.getEntitiesAndMetadata(result.entities, result.compositeEntities, verbose);
+                let recognizerResult  : RecognizerResult = {
+                    text: result.query,
+                    intents: $this.getIntents(result.intents),
+                    entities: entitiesAndMetadata.entities
+                };
+                if(verbose) {
+                    recognizerResult.$instance = {
+                        entities : entitiesAndMetadata.metadata
```

Comment:
For intents is the right Node thing to do, as not have it there or to have intents:[{}, ...] for each intent in intents?  If we did that then you can always do the path for intent meta-data--even if there is none.  If it is reasonable for people to check for the presence of intents, it would be OK to not have it.

Reply:
What would be the "value" for intent in the metadata? The Score again, or something else?

$instance.intents = { 'OrderSandwich': XYZ, 'OrderDrink': KLM }, what's XYZ/KLM? Also intents isn't an array in the main JSON, did you intent to make it an array in the metadata?

Reply:
I opted in for having an empty object placeholder.